app/lib/hooks/useRepoChats.ts

```diff
@@ -0,0 +1,349 @@
+            id: `assistant-${commit.id}-${generateId()}`,
+            role: 'assistant',
+            content: assistantContent,
+            parts: [
+              {
+                type: 'text',
+                text: assistantContent,
+              },
+            ],
+            createdAt: new Date(commit.created_at),
+          });
+        }
+      }
+
+      const userMessageMatch = commit.message.match(/<V8UserMessage>([\s\S]*?)<\/V8UserMessage>/);
+      const userMatched = userMessageMatch && userMessageMatch[1];
+
+      if (userMatched) {
+        const userContent = userMessageMatch[1].trim();
+
+        if (userContent) {
+          messages.push({
+            id: `user-${commit.id}-${generateId()}`,
+            role: 'user',
+            content: userContent,
+            parts: [
+              {
+                type: 'text',
+                text: userContent,
+              },
+            ],
+            createdAt: new Date(commit.created_at),
+          });
+        }
+      }
+
+      if (!assistantMatched && !userMatched && commit.message.toLowerCase().trim() !== 'initial commit') {
+        messages.push({
+          id: `commit-${commit.id}-${generateId()}`,
+          role: 'assistant',
+          content: commit.message,
+          parts: [
+            {
+              type: 'text',
+              text: commit.message,
+            },
+          ],
+          createdAt: new Date(commit.created_at),
+        });
+      }
+    });
+
+    return messages;
+  }, []);
+
+  // Fetch project files
+  const fetchProjectFiles = useCallback(async (projectPath: string): Promise<FileMap> => {
+    setLoadingFiles(true);
+
+    try {
+      const zipBlob = await downloadProjectZip(projectPath);
+
+      // Load zip file using JSZip
+      const zip = await JSZip.loadAsync(zipBlob);
+
+      // Process zip contents into FileMap structure
+      const fileMap: FileMap = {};
+      const dirSet = new Set<string>(); // 디렉토리 경로 추적용 Set
+
+      // 먼저 모든 디렉토리 경로를 수집
+      zip.forEach((relativePath) => {
+        // 경로에서 첫 번째 폴더(프로젝트 루트)를 제거
+        const pathParts = relativePath.split('/');
+
+        if (pathParts.length > 1) {
+          pathParts.shift(); // 첫 번째 경로 부분(프로젝트 폴더) 제거
+        }
+
+        // 파일 경로의 모든 상위 디렉토리를 찾아 dirSet에 추가
+        if (pathParts.length > 1) {
+          for (let i = 1; i < pathParts.length; i++) {
+            const dirPath = pathParts.slice(0, i).join('/');
+
+            if (dirPath) {
+              dirSet.add(dirPath);
+            }
+          }
+        }
+      });
+
+      // 디렉토리 먼저 FileMap에 추가
+      dirSet.forEach((dirPath) => {
+        const fullPath = `${WORK_DIR}/${dirPath}`;
+        fileMap[fullPath] = {
+          type: 'folder',
+        };
+      });
+
+      const promises: Promise<void>[] = [];
+
+      // 파일 처리
+      zip.forEach((relativePath, zipEntry) => {
+        if (!zipEntry.dir) {
+          const promise = zipEntry.async('string').then((content) => {
```

Comment:
How about `await` instead of `.then()` chain? it helps error handling with `finally` block